---
description: MSSQLSERVER_3859
title: MSSQLSERVER_3859
ms.custom: ''
ms.date: 08/20/2020
ms.prod: sql
ms.reviewer: ramakoni1, pijocoder, suresh-kandoth, Masha
ms.technology: supportability
ms.topic: language-reference
helpviewer_keywords:
- 3859 (Database Engine error)
ms.assetid: ''
author: suresh-kandoth
ms.author: ramakoni
ms.openlocfilehash: 3228a21b43fd5b5ed2a23c5e13b47353e55cd028
ms.sourcegitcommit: a9e982e30e458866fcd64374e3458516182d604c
ms.translationtype: HT
ms.contentlocale: ru-RU
ms.lasthandoff: 01/11/2021
ms.locfileid: "98093287"
---
# <a name="mssqlserver_3859"></a>MSSQLSERVER_3859
 [!INCLUDE [SQL Server](../../includes/applies-to-version/sqlserver.md)]

## <a name="details"></a>Сведения

|attribute|Значение|
|---|---|
|Название продукта|SQL Server|
|Идентификатор события|3859|
|Источник события|MSSQLSERVER|
|Компонент|SQLEngine|
|Символическое имя|DBCC_CHECKCAT_DIRECT_UPDATE|
|Текст сообщения|Предупреждение. Системный каталог был явно обновлен в базе данных с идентификатором \%d, последний раз в %S_DATE.|
||

## <a name="explanation"></a>Пояснение

Эта ошибка означает, что пользователь инициировал изменения в системных таблицах. Обновление системных таблиц вручную не поддерживается. Системные таблицы должны обновляться только ядром СУБД [!INCLUDE[ssNoVersion](../../includes/ssnoversion-md.md)]. Когда [!INCLUDE[ssNoVersion](../../includes/ssnoversion-md.md)] обнаруживает изменения, инициированные пользователем в системных таблицах, возникает ошибка 3859 в следующих двух сценариях:

- Сценарий 1

    При запуске базы данных [!INCLUDE[ssNoVersion](../../includes/ssnoversion-md.md)], которая содержит системную таблицу, обновленную вручную, в журнале ошибок [!INCLUDE[ssNoVersion](../../includes/ssnoversion-md.md)] или в журнале приложения в компоненте "Просмотр событий" регистрируется событие наподобие следующего:

    > Имя журнала: Приложение  
    Источник: MSSQLSERVER ИД события: 3859  
    Категория задачи: Сервер  
    Уровень: Сведения  
    Описание: Предупреждение. Системный каталог был явно обновлен в базе данных с идентификатором \%d, последний раз в **date_time**.  

- Сценарий 2  

    При выполнении команды `DBCC_CHECKDB` после обновления системной таблицы вручную возвращается предупреждение наподобие следующего:

    > Результаты выполнения команды DBCC для "**имя_базы_данных**".  
    Сообщение 8992, уровень 16, состояние 1, строка 1  
    Проверьте сообщение каталога 3859, состояние 1: Предупреждение. Системный каталог был явно обновлен в базе данных с идентификатором \%d, последний раз в **date_time**.  
    CHECKDB нашла 0 ошибок размещения и 0 ошибок согласованности в базе данных "**имя_базы_данных**".  
    Выполнение инструкции DBCC завершено. Если инструкция DBCC выдает сообщения об ошибках, обратитесь к системному администратору.

## <a name="user-action"></a>Рекомендуемые действия

Чтобы решить эту проблему, используйте один из указанных ниже способов.

- Метод 1

    При наличии чистой резервной копии базы данных восстановите базу данных из нее.  
    > [!NOTE]
    > Этот метод работает только в том случае, если в резервной копии нет несогласованности метаданных.  

- Метод 2  

    Если восстановить базу данных из резервной копии не удается, экспортируйте данные и объекты в новую базу данных. Затем перенесите содержимое базы данных, обновленной вручную, в новую базу данных. Обратите внимание, что несоответствия в системных каталогах невозможно исправить с помощью параметров восстановления команд DBCC CHECKDB. Так как команда не может исправить поврежденные метаданные, она не предоставляет рекомендуемого уровня восстановления.

    > [!NOTE]
    > Данные из системных таблиц можно просмотреть с помощью представлений системных каталогов.

## <a name="more-information"></a>Дополнительные сведения

Дополнительные сведения см. в следующей статье: [Системные базовые таблицы](../system-tables/system-base-tables.md).